### PR TITLE
DI enabled Microsoft.AspNetCore.Sockets

### PR DESCRIPTION
**Pull request by davidfowl**
- Added AddSockets extension method to IServiceCollection
- Inject IApplicationLifetime into ConnectionManager to handle
graceful shutdown and added test.
- Call AddSockets from AddSignalR

### CODE REVIEW

File: src/Microsoft.AspNetCore.SignalR/SignalRDependencyInjectionExtensions.cs [hunk 1]
```diff
@@ -8,10 +8,11 @@
 
 namespace Microsoft.Extensions.DependencyInjection
 {
-    public static class DependencyInjectionExtensions
+    public static class SignalRDependencyInjectionExtensions
     {
         public static ISignalRBuilder AddSignalR(this IServiceCollection services)
```
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.SignalR/SignalRDependencyInjectionExtensions.cs]: While you're here should we make this more like other projects and not have an `ISignalRBuilder` and make an overload that accepts `Action<SignalROptions>`

File: samples/SocketsSample/Startup.cs [hunk 2]
```diff
@@ -16,18 +16,17 @@ public class Startup
...
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddRouting();
-
             services.AddSingleton<ProtobufInvocationAdapter>();
             services.AddSingleton<LineInvocationAdapter>();
 
-            services.AddSignalR()
-                    .AddSignalROptions(options =>
+            services.AddSockets();
```
Comment (moozzyk) [inline samples/SocketsSample/Startup.cs]: `AddSignalR` adds sockets so this line seems unneeded.

Comment (davidfowl) [inline samples/SocketsSample/Startup.cs]: I wanted to add it explicitly anyways.

Comment (moozzyk) [inline samples/SocketsSample/Startup.cs]: Why? Do you expect people do this? Because they will if they see it in a sample...

Comment (davidfowl) [inline samples/SocketsSample/Startup.cs]: Yes especially if they need to configure options for sockets (which there will inevitably be).

Comment (moozzyk) [inline samples/SocketsSample/Startup.cs]: I see.

**Pull request merged**